test: cleanup attw deps

jason-ha opened the pull request:
are-the-types-wrong checks should not require tests to be built. Reduce requirements to production steps.

Refactor test packs in non-private packages from automatic with `pack` to an explicit package script. Otherwise, those packages will require tests to be built before attw as attw does a pack. (postpack in such cases would fail without the test files present.)

This also prevents unstaged test tar files from showing up after running `check:are-the-types-wrong`.

`examples/benchmark/tablebench` is left packing tests automatically in postpack. (It is private and doesn't use attw.)

jason-ha committed:
test: cleanup attw deps
are-the-types-wrong checks should not require tests to be built. Reduce requirements to production steps.

Refactor test packs from automatic with `pack` to an explicit package script. Otherwise, those packages will require tests to be built before attw as attw does a pack. (postpack in such cases would fail without the test files present.)

This also prevents unstaged test tar files from showing up after running `check:are-the-types-wrong`.

alexvy86 commented on scripts/pack-packages.sh:
```diff
@@ -14,12 +14,19 @@ mkdir $STAGING_PATH/test-files/
 # If test files are found, they are moved to the test-files folder.
+# Note: use of package's pack:tests is only supported for pnpm as PACKAGE_MANAGER.
 if [ -f ".releaseGroup" ]; then
+  if [ "$PACKAGE_MANAGER" == "pnpm" ]; then
+    pnpm -r --if-present pack:tests
+  fi
```
Would it be better to put this below before the third `flub exec`, to avoid any risk of the `.test-files.tar` files ending up inside the tarballs for the actual packages? I see we have entries in `.npmignore` for that (are they even relevant today? Since the tarballs are generated at `postpack`), but if someone adds a test-files tar to a new package and forgets the `.npmignore` bit... And similar concern below.

jason-ha replied:
Those `flub exec` are all one big command line, sequence. I don't know the magic to embed a conditional command inside, but otherwise, yes.
I can *guess* they were done like that so if there is an early failure, then nothing else executes.

alexvy86 reviewed (commented):
Just for future reference: replacing postpack with a script that has to be called manually seems fine. When someone runs `pnpm pack` locally to test something they won't automatically get the `test-files.tar` file like they do today, but in pretty much all cases I don't think they needed it anyway, it's only relevant during CI.

jason-ha committed:
ci: fix: skip example from postpack cleanup
as it is a private package and doesn't have attw that is impacted. So `postpack` is fine.

Also limit general pack:tests to non-private packages as originally intended.

jason-ha commented:
> Just for future reference: replacing postpack with a script that has to be called manually seems fine. When someone runs `pnpm pack` locally to test something they won't automatically get the `test-files.tar` file like they do today, but in pretty much all cases I don't think they needed it anyway, it's only relevant during CI.

That is fully my intent. It seemed like a backdoor to pack tests. Since there are definitely cases that don't want those to be created, then we should make the steps explicit. We could add in a pack-with-tests script or something else if there is a need for that.
